util/DirectorySourceQueue.cpp

```diff
@@ -360,7 +366,9 @@ bool DirectorySourceQueue::explore() {
               !std::regex_match(newRelativePath, includeRegex)) {
             continue;
           }
-          WdtFileInfo fileInfo(newRelativePath, fileStat.st_size, directReads_);
+          int perm = fileStat.st_mode & (S_IRWXU | S_IRWXG | S_IRWXO);
```

Comment:
Oh my previous understanding was: we allow the user to specify a permission, which will be used during createFile. If no valid permission is specified in the request, we will create a 644. (That's why my default permission value is 512. Cuz 0 is still a theoretically valid permission) But I think you are right. It is better to always read the permission from the given file.